### PR TITLE
convert scoreboard.js to an ES6 class

### PR DESCRIPTION
**Pull request by 1by**

### CODE REVIEW

File: lib/scoreboard.js [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-  return this.itemsMap[name]
-}
 
-ScoreBoard.prototype.remove = function (name) {
-  const removed = this.itemsMap[name]
-  delete this.itemsMap[name]
-  return removed
-}
 
 Object.defineProperty(ScoreBoard.prototype, 'items', {
```
Comment (wvffle) [inline lib/scoreboard.js]: You can use 
```js
get items () {
  return this[2]
}
```

File: lib/scoreboard.js [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
+class ScoreBoard {
+  constructor(packet) {
+    this.name = packet.name
+    this.title = packet.displayText
+    this.itemsMap = {}
+  }
+  add(name, value) {
+    this.itemsMap[name] = { name, value }
+    return this.itemsMap[name]
+  }
```
Comment (wvffle) [inline lib/scoreboard.js]: \n

File: lib/scoreboard.js [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
-function ScoreBoard (packet) {
-  this.name = packet.name
-  this.title = packet.displayText
-  this.itemsMap = {}
+class ScoreBoard {
+  constructor(packet) {
+    this.name = packet.name
+    this.title = packet.displayText
+    this.itemsMap = {}
+  }
```
Comment (wvffle) [inline lib/scoreboard.js]: \n

File: lib/scoreboard.js [hunk 4]
```diff
@@ -1,44 +1,43 @@
-module.exports = ScoreBoard
-
 const sortItems = (a, b) => {
   if (a.value > b.value) return -1
   if (a.value < b.value) return 1
   return 1
 }
```
Comment (wvffle) [inline lib/scoreboard.js]: \n

File: lib/scoreboard.js [hunk 5]
```diff
@@ -21,12 +21,10 @@ class ScoreBoard {
     delete this.itemsMap[name]
     return removed
   }
-}
-Object.defineProperty(ScoreBoard.prototype, 'items', {
   get() {
```
Comment (wvffle) [inline lib/scoreboard.js]: lacking name

**Comment by wvffle** [comment]
Still, add newlines

**Comment by 1by** [comment]
Trying to learn GitHub so if the pr is messy im sorry :joy:

**Comment by wvffle** [comment]
> Trying to learn GitHub so if the pr is messy im sorry joy

No problem, just keep in mind that your code should be readable ;)

File: lib/scoreboard.js [hunk 6]
```diff
@@ -4,38 +4,44 @@ const sortItems = (a, b) => {
...
+
+  remove (name) {
+
     const removed = this.itemsMap[name]
     delete this.itemsMap[name]
     return removed
   }
-  get() {
+
+  get () {
```
Comment (wvffle) [inline lib/scoreboard.js]: This should be a getter of `items`, right now it's a `get` method

Comment (1by) [reply lib/scoreboard.js]: right, fixed

**Comment by 1by** [comment]
@wvffle done! Think I understand what u meant.

File: lib/scoreboard.js [hunk 7]
```diff
@@ -11,22 +12,19 @@ class ScoreBoard {
   }
 
   add (name, value) {
-
```
Comment (wvffle) [inline lib/scoreboard.js:8]: Unneeded newline

File: lib/scoreboard.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 const sortItems = (a, b) => {
   if (a.value > b.value) return -1
   if (a.value < b.value) return 1
+
```
Comment (wvffle) [inline lib/scoreboard.js:4]: Unneeded newline, add it between end of a function and class declaration

**Comment by 1by** [comment]
Fixed!

**Pull request merged**